content/blog/2022/2022-08-30-handling-timezones-in-spring.md

```diff
@@ -0,0 +1,516 @@
+**When zone information needs to be saved in the database this format is preferred as it would always represent the same instant on the timeline** (especially when the server and database represent different timezones, conversion that represents time at the same instant would be required).  
+
+Sample conversion example:
+
+````java
+        OffsetDateTime currentDateTime = OffsetDateTime.now();
+        System.out.println("System default timezone current zone offset date/time : " + currentDateTime);
+
+        // Check for format difference
+        ZonedDateTime currentZoneDateTime = ZonedDateTime.now();
+        System.out.println("Current system zone date/time : " + currentZoneDateTime);
+
+        ZoneOffset zoneOffSet= ZoneOffset.of("+01:00");
+        OffsetDateTime offsetDateTime = OffsetDateTime.now(zoneOffSet);
+        System.out.println("Europe/London zone offset date/time : " + offsetDateTime);
+
+        OffsetDateTime fromLocals = OffsetDateTime.of(LocalDate.now(), LocalTime.now(), currentDateTime.getOffset());
+        System.out.println("Get Offset date/time from Locals : " + fromLocals);
+
+        OffsetDateTime fromLocalDateTime = OffsetDateTime.of(LocalDateTime.of(2022, Month.NOVEMBER, 1, 10, 10, 10), currentDateTime.getOffset());
+        System.out.println("Get Offset date/time from LocalDateTime with Offset at the current Instant considered " +
+                "(does not consider DST at custom date): " + fromLocalDateTime);
+
+        OffsetDateTime fromLocalsWithDefinedOffset = OffsetDateTime.of(LocalDate.now(), LocalTime.now(), ZoneOffset.systemDefault().getRules().getOffset(LocalDateTime.of(2022, Month.NOVEMBER, 1, 10, 10, 10)));
+        System.out.println("Get Offset date/time from Local with offset for custom LocalDateTime considered " +
+                "(Considers DST at custom date) : " + fromLocalsWithDefinedOffset);
+
+        OffsetDateTime sameInstantDiffOffset = currentDateTime.withOffsetSameInstant(ZoneOffset.of("+01:00"));
+        System.out.println("Same instant at a different offset : " + sameInstantDiffOffset);
+
+        OffsetDateTime dt = OffsetDateTime.parse("2011-12-03T10:15:30+01:00", DateTimeFormatter.ISO_OFFSET_DATE_TIME);
+        DateTimeFormatter fmt = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss'Z'");
+        System.out.println("OffsetDateTime parsed and formatted" + fmt.format(dt));
+
+        OffsetDateTime convertFromZoneToOffset = currentZoneDateTime.toOffsetDateTime();
+        System.out.println("Convert from ZonedDateTime to OffsetDateTime : " + convertFromZoneToOffset);
+
+````
+
+Output:
+
+{{% image alt="settings" src="images/posts/handling-timezones-in-spring/offsetDateTimeOutput.JPG" %}}
+
+## Compatibility with the legacy API
+
+As a part of the Date/Time API, **methods have been introduced to convert from legacy classes to the newer API objects**.
+
+Sample Code Conversion:
+
+````java
+
+    @Test
+    public void testWorkingWithLegacyDateInJava8() {
```

Comment:
If you use tests in code examples, please include the class so that readers can find the code easier in the GitHub repo. i.e:

```
public class MyTest {
...
  @Test
    public void testWorkingWithLegacyDateInJava8() {
    }
}```

You can leave out any test methods that are not relevant in the current example